addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js

```diff
@@ -3393,11 +3393,11 @@ X[]
                     const anchorOffset = selection.anchorOffset;
                     const p = editor.editable.querySelector('p');
                     const textNode = p.childNodes[0];
-                    triggerEvent(editor.editable, 'keydown', {key: ' ', code: 'Space'});
+                    triggerEvent(editor.editable, 'input', {data: ' ', inputType: 'insertText' });
                     textNode.textContent = "a http://test.com b http://test.com\u00a0 c http://test.com d";
                     selection.extend(textNode, anchorOffset + 1);
                     selection.collapseToEnd();
-                    triggerEvent(editor.editable, 'input', {data: ' ', inputType: 'insertText' });
+                    triggerEvent(editor.editable, 'keydown', {key: ' ', code: 'Space'});
```

Comment:
In both tests I exchanged the 'input' and 'keydown' events since the insertion of the link was moved from the former to the latter. This works properly.
However this test fails because actual != expected:
```
actual  : '<p>a http://test.com b <a href="http://test.com">http://test.com</a>[]&nbsp; c http://test.com d</p>'
expected: '<p>a http://test.com b <a href="http://test.com">http://test.com</a>&nbsp;[] c http://test.com d</p>'
```
It seems that `[]` represents the cursor position which indeed is off for that test. However I don't get at all why the only trigger where we expect a link to be created (6 triggers in two tests, only one is supposed to trigger the link creation) inserts a `\u00a0` (or `&nbsp;`) after the link and expects the link to be created, but without that symbol, but the cursor to be placed after that symbol.
I could fix the selection but it would be adding more code (cf extra `pop()`) whose only purpose is to green tests that I don't understand.

Reply:
@Goaman following your comment I prepared two stashes to revert the event order to what it was:
- one simply moving the `input` event between `keydown` and `keyup`
- another also adapting `contentBefore` and removing the assignment to `textNode.textContent` (and its subsequent adaptation of the selection) as anyway it no longer makes sense to change the content after the event that actually inserts the link.

Both these changes have no impact on the tests, meaning the same test fails in the same way.

I also prepared a stash that moves the handling of `Space` key back to `keydown`, and this does _not_ fix the test either (same error). 

So the issue really comes from the range restoring code provided by @dmo-odoo, see comment above.